container/docker/factory.go

```diff
@@ -49,6 +49,8 @@ var dockerCgroupRegexp = regexp.MustCompile(`.+-([a-z0-9]{64})\.scope$`)
 
 var noSystemd = flag.Bool("nosystemd", false, "Explicitly disable systemd support for Docker containers")
 
+var dockerMetadataEnvs = flag.String("docker_metadata_env", "", "Comma seperated list with names of env variables, which will be exported as metadata (default: empty)")
```

Comment:
this is to select the env that we want to publish as metrics labels ( I made a similar flag for logging in docker https://github.com/docker/docker/pull/16961 )

For example, a mesos container has the following configs:
```
"Env": [
            "GOMAXPROCS=64",
            "MARATHON_APP_VERSION=2015-11-18T22:55:53.442Z",
            "MARATHON_APP_DOCKER_IMAGE=docker.cfops.it/elssftp:2015.10.17",
            "MESOS_TASK_ID=pdx_elsent_svc_elssftp.3f90dfce-9231-11e5-81c8-2c600c83f52f",
            "MARATHON_APP_ID=/pdx/elsent/svc/elssftp",
            "MESOS_SANDBOX=/mnt/mesos/sandbox",
            "MESOS_CONTAINER_NAME=mesos-d311c627-c678-4651-9345-3b46a02c7d99-S115.9961fba3-3680-47d4-ba5a-2b724aa46515",
            "PATH=/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin",
            "container=docker"
],
"Labels": {}, // sad :(
```

we only interest in MARATHON_APP_ID,MESOS_TASK_ID,MESOS_CONTAINER_NAME as the container identifiers.

A flag is reasonably useful for use as we only use 1 orchestrator i.e. mesos, so all containers will have the same set of env keys. This is the chosen approach in the original PR.

I can imagine an alternative approach would be to let the administrator defined their desired env keys as a well known label, for example:

```
// this will not be set as metrics in prometheus
"Labels": { "cadvisor.metadata_env": "MARATHON_APP_ID,MESOS_TASK_ID,MESOS_CONTAINER_NAME" }
```